client/my-sites/stats/stats-notices/index.tsx

```diff
@@ -93,9 +95,11 @@ const NewStatsNotices = ( { siteId, isOdysseyStats, statsPurchaseSuccess }: Stat
 
 	const { isLoading, isError, data: serverNoticesVisibility } = useNoticesVisibilityQuery( siteId );
 
+	// TODO: Integrate checking purchases and plans loaded state into `hasSiteProductJetpackStatsPaid`.
 	const hasLoadedPurchases = useSelector( ( state ) => hasLoadedSitePurchasesFromServer( state ) );
+	const hasLoadedPlans = useSelector( ( state ) => hasLoadedSitePlansFromServer( state, siteId ) );
 
-	if ( ! hasLoadedPurchases || isLoading || isError ) {
+	if ( ! hasLoadedPurchases || ! hasLoadedPlans || isLoading || isError ) {
```

Comment:
I'm wondering how this works for Odyssey - will it load successfully?

Reply:
Nice catch! Fixed in https://github.com/Automattic/wp-calypso/pull/81514/commits/06fc681d4c7ae80e79d2bfcadeda6159744a0ace.